### ISSUE TITLE
False positive `B030` when using `*exc_types or ValueError`

### ISSUE THREAD

**Issue by jamesbraza** [charliermarsh/ruff#4465]
`ruff==0.0.265` has a false positive `B030`:

```python
def foo(msg: str, *exc_types: type[Exception]) -> None:
    try:
        raise ValueError(msg)
    except exc_types or ValueError as exc:
        print(exc)


foo("BOOM")
foo("BOOM2", ValueError)
foo("BOOM3", ValueError, IndexError)
```

```bash
> ruff --select=B030 a.py
a.py:4:12: B030 `except` handlers should only be exception classes or tuples of exception classes
Found 1 error.
```

**Comment by tjkuson** [comment]
I am not sure this is a false positive, seeing as, to my understanding, binary operations shouldn't be used in `except` clauses, as only the first exception will be caught.

Doing this instead passes the check and should work better.

```python
def foo(msg: str, *exc_types: type[Exception]) -> None:
    try:
        raise ValueError(msg)
    except (*exc_types, ValueError) as exc:
        print(exc)


foo("BOOM")
foo("BOOM2", ValueError)
foo("BOOM3", ValueError, IndexError)
```

**Comment by charliermarsh** [comment]
Yeah this looks like a true positive to me. What behavior are you seeing, and what are you expecting?

**Comment by charliermarsh** [comment]
Ahhh I see! Thanks for sharing. That makes sense.

I see how this could be considered a false positive, but to be honest, I think this is the rule working as intended. The pattern that's being used here, even if it has the correct runtime behavior, is confusing -- as evidence, the two readers on this issue thought the intended behavior was different from the author's intended behavior. I'd argue that it's correct for the rule to flag this case, even if the runtime behavior is correct.

My humble suggestion would be to assign `exc_types or ValueError` to a temporary variable and match on that, which IMO is clearer.
